test/opml.cpp

```diff
@@ -0,0 +1,312 @@
+#include "opml.h"
+
+#include <libxml/xmlsave.h>
+#include "cache.h"
+#include "file_urlreader.h"
+
+#include "3rd-party/catch.hpp"
+#include "test-helpers.h"
+
+using namespace newsboat;
+
+TEST_CASE("prepare_opml creates an XML document with feed URLs in OPML format",
+		"[opml]")
+{
+	const auto check =
+		[]
+		(const FeedContainer& feedcontainer)
+		-> std::string
+		{
+			xmlDocPtr opml = OPML::prepare_opml(feedcontainer);
+
+			xmlBufferPtr buffer = xmlBufferCreate();
+
+			xmlSaveCtxtPtr context = xmlSaveToBuffer(buffer, nullptr, 0);
+			xmlSaveDoc(context, opml);
+			xmlSaveClose(context);
+
+			const std::string opmlText(
+					reinterpret_cast<char*>(buffer->content),
+					buffer->use);
+
+			xmlBufferFree(buffer);
+			buffer = nullptr;
+
+			return opmlText;
+		};
+
+	SECTION("No feeds") {
+		FeedContainer feeds;
+
+		const std::string expectedOpmlText(
+			"<?xml version=\"1.0\"?>\n"
+			"<opml version=\"1.0\">"
+			"<head><title>newsboat - Exported Feeds</title></head>"
+			"<body/></opml>\n");
+
+		REQUIRE(check(feeds) == expectedOpmlText);
+	}
+
+	SECTION("A few feeds") {
+		configcontainer cfg;
+		cache rsscache(":memory:", &cfg);
+		FeedContainer feeds;
+
+		std::shared_ptr<rss_feed> feed
+			= std::make_shared<rss_feed>(&rsscache);
```

Comment:
Other code segments have the `=` in the line above. Not sure, whether we are free to choose or our style guide wants us to do it a specific way.

Reply:
The style guide doesn't specify. I'll move `=` to the first line for consistency.